### ISSUE TITLE
How to fix build errors on a iOS project?

### ISSUE THREAD

**Issue by KosukeOhmura** [grpc/grpc-swift#100]
Hi,
I tried to import grpc-swift to a brand new iOS project on XCode 9.2 working on macOS 10.13.2, but I encountered some build errors and I can't fix it. 

I did:
* create a new iOS project 
* import [apple/swift-protobuf](https://github.com/apple/swift-protobuf) via Carthage
* add SwiftGRPC.xcodeproj from "Add Files to ..." of menu bar
* add BoringSSL, CgRPC, gRPC, and CzLib from 'Linked Frameworks and Libraries' of General tab

Can you tell me what I am wrong please?

project: https://github.com/KosukeOhmura/grpc_practice_ios

Error logs:
```
zlib-example
Check dependencies
target specifies product type 'com.apple.product-type.tool', but there's no such product type for the 'iphonesimulator' platform

Czlib
clang: error: no such file or directory: '/Users/ohmurakosuke/Development/grpc_practice_ios/GrpcSample/.build/checkouts/zlib.git--9016602441419864029/Sources/Czlib/adler32.c'
clang: error: no input files
clang: error: no such file or directory: '/Users/ohmurakosuke/Development/grpc_practice_ios/GrpcSample/.build/checkouts/zlib.git--9016602441419864029/Sources/Czlib/compress.c'
clang: error: no input files
```

Thanks,

**Comment by bilalyaylak** [comment]
grpc-swift currently has no Carthage support.  
If you want, i forked the project recently (0.3.0) and generated a xcode project compatible with Carthage and iOS 8.0+  
https://github.com/bilalyaylak/grpc-swift  
Maintaining Carthage support for this project is diffucult because of Swift Package Manager. Maybe, some ruby script can be written to edit/delete targets of the xcodeproject which is generated by Swift Package Manager.

**Comment by KosukeOhmura** [comment]
@bilalyaylak 
Thanks for your explaining.
I didn't use Carthage for grpc-swift. Do you mean I can't use Carthage for swift-protobuf too?

I found `.build` directory is generated with SwiftGRPC.xcodeproj by `make` command. It includes clang files that Czlib's error says. Can I copy `.build` to my project directory for fixing Czlib's error?

Thanks,

**Comment by bilalyaylak** [comment]
Sorry, I misread the issue. zlib-example cannot be compiled for iOS, you can delete zlib-example target when you want to use grpc-swift on iOS project. 
By the way, you should also add CgRPC headers directory to your "Header Search Paths"  
```
$(SRCROOT)/Carthage/Checkouts/grpc-swift/Sources/CgRPC/include
```  
I believe that doing so may solve Czlib related problems.  I cannot remember if there were another problematic case. You can keep only framework targets.

**Comment by KosukeOhmura** [comment]
@bilalyaylak 
Thank you, I succeeded to build it. I did:

* delete zlib-example from TARGETS
* copy `.build` and `Sources` directory into my project directory

I didn't add CgRPC headers by your way. There is no directory `$(SRCROOT)/Carthage/Checkouts/grpc-swift` because I did not use Carthage for grpc-swift.

Copying directories manually is not a smart way, but anyway I succeeded to build it so I close this issue.
Thanks.
